src/components/MenuPanel/MenuPanel.js

```diff
@@ -201,19 +200,31 @@ class MenuPanel extends React.PureComponent {
                       {systemApps.map(app => this.renderAppGroup(app))}
                     </AnimDiv>
                   </div>
-                  <Spring
-                    config={springs.smooth}
-                    from={{ opacity: 0 }}
-                    to={{
-                      opacity: activeInstanceId === 'console',
-                    }}
-                  >
-                    {props => (
-                      <div style={props}>
-                        {this.renderAppGroup(APP_CONSOLE)}
-                      </div>
-                    )}
-                  </Spring>
+                  <div>
+                    <ConsoleOptOutContext.Consumer>
+                      {({ optedOut }) =>
+                        !optedOut ? (
```

Comment:
Not sure if there's a way to avoid using this ternary here, by using a prop on the spring, but perhaps @bpierre knows.

Reply:
Yes! Using `immediate` should work:

```jsx
<ConsoleOptOutContext.Consumer>
  {({ optedOut }) => (
    <Spring
      immediate={!optedOut}
      config={springs.swift}
      from={{ opacity: 0 }}
      to={{
        opacity: Number(window.location.hash.includes('console') && optedOut),
      }}
    >
      {({ opacity }) => (
        <div style={{ opacity }}>{this.renderAppGroup(APP_CONSOLE)}</div>
      )}
    </Spring>
  )}
</ConsoleOptOutContext.Consumer>
```

Some other suggestions:

- I would use `springs.swift` here, since it’s a motionless transition on a small object.
- I think react-spring was ok with a boolean after `from` contains a number, but I would still convert `opacity` to a `Number` in `to`, just to be explicit.
- I haven’t done it in the example, but we should probably avoid referring to `window.location` from here. I think `activeInstanceId` must be set to `console` already?

Reply:
I see! much cleaner this way! 🔥